src/app/Plans/components/helpers.ts

```diff
@@ -1,122 +1,175 @@
-      condition.type === PlanStatusType.Failed
+      condition.type === 'Ready' ||
+      condition.type === 'Executing' ||
+      condition.type === 'Succeeded' ||
+      condition.type === 'Failed'
   );
-  return condition ? PlanStatusDisplayType[condition.type] : '';
+  return condition ? condition.type : '';
 };
 
-// TODO maybe generalize this for cold migrations too
-type WarmPlanState =
-  | 'NotStarted'
-  | 'Starting'
-  | 'Copying'
-  | 'AbortedCopying'
-  | 'StartingCutover'
-  | 'Cutover'
-  | 'Finished';
-
-export const getWarmPlanState = (
+export const getMigStatusState = (state: PlanState | null, isWarmPlan: boolean) => {
+  let title = '';
+  let variant: ProgressVariant | undefined;
+
+  switch (true) {
+    case state === 'Starting': {
+      title = 'Running';
+      break;
+    }
+    case state === 'NotStarted': {
+      title = '';
+      break;
+    }
+    case state === 'Finished-Failed':
+    case state === 'AbortedCopying': {
+      title = 'Failed';
+      variant = ProgressVariant.danger;
+      break;
+    }
+    case state === 'Finished-Succeeded': {
+      title = 'Succeeded';
+      variant = ProgressVariant.success;
+      break;
+    }
+    case state === 'Copying':
+    case state === 'PipelineRunning': {
+      title = isWarmPlan ? 'Running cutover' : 'Running';
+      break;
+    }
+    case state === 'Finished-Incomplete': {
+      title = 'Finished - Incomplete';
+      variant = ProgressVariant.warning;
+      break;
+    }
+    default: {
+      title = '';
+    }
+  }
+
+  return {
+    title,
+    variant,
+  };
+};
+
+export const getButtonState = (state: PlanState | null): PlanActionButtonType | null => {
+  let type: PlanActionButtonType | null = null;
+
+  switch (true) {
+    case state === 'NotStarted': {
+      type = 'Start';
+      break;
+    }
+    case state === 'Starting': {
+      type = null;
+      break;
+    }
+    case state === 'Copying': {
+      type = 'Cutover';
+      break;
+    }
+    case state === 'Finished-Incomplete':
+    case state === 'Finished-Failed':
+    case state === 'Canceled':
+    case state === 'AbortedCopying': {
+      type = 'Restart';
+      break;
+    }
+    case state === 'StartingCutover': {
+      type = null;
+      break;
+    }
+    case state === 'PipelineRunning': {
+      type = null;
+      break;
+    }
+    case state === 'Finished-Succeeded': {
+      type = null;
+      break;
+    }
+    default: {
+      type = null;
+    }
+  }
+
+  return type;
+};
+
+export const getPlanState = (
   plan: IPlan | null,
-  migration: IMigration | null,
-  migrationsQuery: UseQueryResult<IKubeList<IMigration>>
-): WarmPlanState | null => {
-  if (!plan || !plan.spec.warm) return null;
+  migration: IMigration | null
+): PlanState | null => {
+  if (!plan) return null;
+
   if (!migration) return 'NotStarted';
-  if (isPlanBeingStarted(plan, migration, migrationsQuery)) {
-    return 'Starting';
-  }
+
```

Comment:
To match my above comment, this should be `if (!migration || !plan.status?.migration?.started) return 'NotStarted';`

Also, it doesn't look like this function ever returns the `'Starting'` state. I see you deleted the `isPlanBeingStarted` function, I think you still need that code, just as a private helper that is only called in `getPlanState`. That probably belongs as an `if(isPlanBeingStarted(plan)) return 'Starting'` here after the NotStarted case.